azure-iot-device/azure/iot/device/provisioning/sk_provisioning_device_client.py

```diff
@@ -0,0 +1,87 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+This module contains one of the implementations of the Provisioning Device Client which uses Symmetric Key authentication.
+"""
+import logging
+from threading import Event
+import traceback
+from .provisioning_device_client import ProvisioningDeviceClient
+from .internal.polling_machine import PollingMachine
+
+logger = logging.getLogger(__name__)
+
+
+class SymmetricKeyProvisioningDeviceClient(ProvisioningDeviceClient):
+    """
+    Client which can be used to run the registration of a device with provisioning service
+    using Symmetric Key authentication.
+    """
+
+    def __init__(self, mqtt_state_based_provider):
+        """
+        Initializer for the Symmetric Key Registration Client
+        :param mqtt_state_based_provider: The state-based protocol provider. As of now this only supports MQTT.
+        """
+        super(SymmetricKeyProvisioningDeviceClient, self).__init__(mqtt_state_based_provider)
+        self._polling_machine = PollingMachine(mqtt_state_based_provider)
+        # To be defined by sample
+        self.on_registration_complete = None
```

Comment:
>self.on_registration_complete = None [](start = 8, length = 36)

If this is called only when the register operation completes, it should be a callback parameter on the register function.

Reply:
without this the registration will work as well..are you meaning to say that the user can call with a [assed in handler

---
In reply to: [282688988](https://github.com/Azure/azure-iot-sdk-python-preview/pull/81#discussion_r282688988) [](ancestors = 282688988)